Documentation/upgrade-patch.md

```diff
@@ -0,0 +1,93 @@
+
+In this guide, we will be upgrading a live Rook cluster running `v0.8.0` to version `v0.8.1`.
+
+### Operator
+The operator controls upgrading all the components and is generally the first component to be updated.
+The operator can be updated by setting the deployment version.
+
+```bash
+kubectl -n rook-ceph-system set image deploy/rook-ceph-operator rook-ceph-operator=rook/ceph:v0.8.1
+```
+
+The operator pod will automatically be restarted by Kubernetes with the new version.
+
+Once you've verified the operator is `Running` and on the new version, verify the health of the cluster is still OK.
+Instructions for verifying cluster health can be found in the [health verification section](#health-verification).
+
+### System Daemons
+The pods in the rook-ceph-system namespace will all be updated automatically when the operator is updated. After the operator is updated, you will see the `rook-ceph-agent` and `rook-discover` pods restarted on the new version.
+
+### Monitors
+There are multiple monitor pods to upgrade and they are each individually managed by their own replica set.
+**For each** monitor's replica set, you will need to update the pod template spec's image version field to `rook/ceph:v0.8.1`.
+For example, we can update the replica set for `mon0` with:
+```bash
+kubectl -n rook-ceph set image replicaset/rook-ceph-mon0 rook-ceph-mon=rook/ceph:v0.8.1
+```
+
+Once the replica set has been updated, we need to manually terminate the old pod which will trigger the replica set to create a new pod using the new version.
+```bash
+kubectl -n rook-ceph delete pod -l mon=rook-ceph-mon0
+```
+
+At this point, it's very important to ensure that all monitors are `OK` and `in quorum`.
+Refer to the [status output section](upgrade.md#status-output) for instructions.
+If all of the monitors (and the cluster health overall) look good, then we can move on and repeat the same upgrade steps for the next monitor until all are completed.
+
+### Object Storage Daemons (OSDs)
+The automatic upgrade of OSD pods has been implemented by the operator. Within a minute of starting the operator on the new version, you should see the osd pods automatically started on the new version.
+Going forward, if there is any change needed to the OSD deployment as determined by the operator, the OSD pods will automatically be updated and restarted.
+For example, the OSDs will be automatically updated when:
+- The version of the operator container changes
+- The `resources` or `placement` elements are changed in the cluster CRD
+
+One by one, as each of the OSDs are updated the operator will wait for the OSD pod to be running again before continuing with the next OSD.
+In some scenarios, the operator will need to be restarted in order to apply the changes to the OSD deployment specs.
```

Comment:
isn't an operator restart required for ALL scenarios except for node add/remove and dashboard enabled/disabled?  Those are the only scenarios supported in `clusterChanged`: https://github.com/rook/rook/blob/release-0.8/pkg/operator/ceph/cluster/cluster.go#L166